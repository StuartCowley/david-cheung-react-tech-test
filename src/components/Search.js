import React, { useState } from "react";
import getImages from "../requests/getImages.js";
import PropTypes from "prop-types";
import "../styles/search.css";

const Search = ({ setFetchData, setErrorMessage, setSlideOn }) => {
  const [value, setValue] = useState();

  const handleSubmit = async (event) => {
    event.preventDefault();

    setSlideOn(false);

    const results = await getImages(value);
    if (results.length === 0) {
      setErrorMessage("Image not found, please try again!");
    } else {
      setErrorMessage("");
    }
    setFetchData(results);
  };

  return (
    <div className="search">
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          onChange={(e) => setValue(e.target.value)}
        />
        <button className="search-button" type="submit">
          Go!
        </button>
      </form>
    </div>
  );
};

export default Search;

Search.propTypes = {
  setFetchData: PropTypes.func.isRequired,
  setErrorMessage: PropTypes.func.isRequired,
  setSlideOn: PropTypes.func.isRequired,
};
