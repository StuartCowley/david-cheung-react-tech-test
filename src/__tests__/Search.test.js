import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Search from "../components/Search";

describe("Search", () => {
  const validProps = {
    setFetchData: jest.fn(),
    setErrorMessage: jest.fn(),
    setSlideOn: jest.fn(),
  };

  test("Renders as expected", () => {
    const rendered = renderer.create(
      <Search
        setFetchData={validProps.setFetchData}
        setErrorMessage={validProps.setErrorMessage}
        setSlideOn={validProps.setSlideOn}
      />
    );

    expect(rendered).toMatchSnapshot();
  });

  test("Assert search-input is present", () => {
    render(
      <Search
        setFetchData={validProps.setFetchData}
        setErrorMessage={validProps.setErrorMessage}
        setSlideOn={validProps.setSlideOn}
      />
    );

    expect(screen.getByRole("textbox")).toHaveClass("search-input");
  });

  test("Assert one button is present", () => {
    render(
      <Search
        setFetchData={validProps.setFetchData}
        setErrorMessage={validProps.setErrorMessage}
        setSlideOn={validProps.setSlideOn}
      />
    );
    const buttons = screen.getAllByRole("button");

    expect(buttons).toHaveLength(1);
    expect(buttons[0]).toHaveTextContent("Go!");
  });
});
