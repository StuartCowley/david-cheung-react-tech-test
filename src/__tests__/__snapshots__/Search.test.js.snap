// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Renders as expected 1`] = `
<div
  className="search"
>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <input
      className="search-input"
      onChange={[Function]}
      type="text"
    />
    <button
      className="search-button"
      type="submit"
    >
      Go!
    </button>
  </form>
</div>
`;
